api/opds.py

```diff
@@ -1722,7 +1723,17 @@ def single_item_feed(
             )
 
         _db = Session.object_session(item)
-        work = license_pool.work or license_pool.presentation_edition.work
+
+        # Sometimes the pool or work may be None
+        # In those cases we have to protect against the exceptions
+        try:
+            work = license_pool.work or license_pool.presentation_edition.work
+        except AttributeError:
+            return NOT_FOUND_ON_REMOTE
+
+        if not work:
+            return NOT_FOUND_ON_REMOTE
```

Comment:
Minor, because this looks like it would work fine: Instead of using both a try/except and the conditional, could we just check all these conditions at once and then either return the problem detail or set `work`?

Reply:
This was my original approach but it ends being a double nested and/or statement which is confusing to understand and hence would be confusing to debug in the future.